### PR TITLE
fix: remove "Error when determining async func"

### LINKED ISSUE #12639: [Bug]: Unexpected error in console due to current `isFunctionAsync` check
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2022-04-06 at 11 14 55 PM

### Steps To Reproduce

1. Create an app with a Filepicker widget and an API name `Api1`
2. Create a JS Object with the code mentioned below
```
export default {
	myFun2:() => {
		console.log(FilePicker1.files);
		Api1.run().then(()=>{
      showAlert("hello");
    })
	}
}
```
3. Refresh the page and notice the dev tools console log

### Public Sample App

https://release.app.appsmith.com/console-errors-isasync/page1-624daa8fdbe66e13e8d84ad4

### Version

1.6.17

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

When Appsmith parse function internally to determine if a function is `async` or not, when that function faces an error we used to log that error to console.

This was confusing for our users who thought due to this error that something went wrong.
hence we remove "Error when determining async func". 

Fixes #12639

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/consoleErrors-async 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/workers/evaluate.ts | 85.62 **(0.1)** | 53.95 **(-0.1)** | 84 **(0)** | 84.67 **(0.11)**</details>

### CODE REVIEW

**Review by ohansFavour — APPROVED** [review]

File: app/client/src/workers/evaluate.ts [hunk 1]
```diff
@@ -415,7 +415,8 @@ export function isFunctionAsync(
         }
       }
     } catch (e) {
-      console.error("Error when determining async function", e);
+      // We do not want to throw errors for internal operations
+      // console.error("Error when determining async function", e);
```
Comment (mohanarpit) [inline app/client/src/workers/evaluate.ts:419]: Nit: If this log message is not required, can we remove the code instead of commenting it out.

Comment (Rishabh-Rathod) [reply app/client/src/workers/evaluate.ts]: sure. makes sense.

**Review by eco-monk — DISMISSED** [review]
An alternate to this:
Please consider adding `logLevel` to disable this by default for the user.
So we have a way to enable and investigate any legitimate errors thrown by this.

**Review by eco-monk — COMMENTED** [review]
An alternate to this:
Please consider adding logLevel to disable this by default for the user.
So we have a way to enable and investigate any legitimate errors thrown by this.

**Review by eco-monk — DISMISSED** [review]

**Review by eco-monk — DISMISSED** [review]

**Review by eco-monk — APPROVED** [review]

**Pull request merged**